packages/core/helper-plugin/lib/src/hooks/useAPIErrorHandler/utils/normalizeAPIError/index.js

```diff
@@ -0,0 +1,43 @@
+const ERROR_PREFIX = 'apiError.';
+
+function getPrefixedId(message, callback) {
+  const prefixedMessage = `${ERROR_PREFIX}${message}`;
+
+  // if a prefix function has been passed in it is used to
+  // prefix the id, e.g. to allow an error message to be
+  // set only for a localization namespace
+  if (callback) {
+    return callback(prefixedMessage);
+  }
+
+  return prefixedMessage;
+}
+
+function normalizeError(error, { name, intlMessagePrefixCallback }) {
+  const { message, path } = error;
+
+  return {
+    id: getPrefixedId(message, intlMessagePrefixCallback),
+    defaultMessage: message,
+    name: error?.name ?? name,
+    values: {
+      path: path?.join('.'),
+    },
+  };
+}
+
+export function normalizeAPIError(apiError, intlMessagePrefixCallback) {
```

Comment:
Since we're exposing this as a public API think we should add some docs for this :)

Reply:
incl. the difference between formatting and normalizing 🤔

Reply:
Sure, will do. Do you think we should do it as part of this PR is separately? Given the number of issues this would solve I'd be more for merge early and start writing docs at the same time.

Regarding the question: my idea is that we normalize every API error from the start so that every utility in FE knows what they can expect when they have to deal with an API error. I've went for `values` because that is the `react-intl` API and it would allow us to enrich the error object in a consistent way (e.g. by exposing the regex-pattern to an error message).